linker/shared/src/test/scala/org/scalajs/linker/AnalyzerTest.scala

```diff
@@ -874,6 +874,116 @@ class AnalyzerTest {
     )
     Future.sequence(results)
   }
+
+  @Test
+  def linkTimeIfReachable(): AsyncResult = await {
+    val mainMethodName = m("main", Nil, IntRef)
+    val fooMethodName = m("foo", Nil, IntRef)
+    val barMethodName = m("bar", Nil, IntRef)
+
+    val thisType = ClassType("A", nullable = false)
+
+    val productionMode = true
+
+    /* linkTimeIf(productionMode) {
+     *   this.foo()
+     * } {
+     *   this.bar()
+     * }
+     */
+    val mainBody = LinkTimeIf(
+      BinaryOp(BinaryOp.Boolean_==,
+          LinkTimeProperty("core/productionMode")(BooleanType),
+          BooleanLiteral(productionMode)),
+      Apply(EAF, This()(thisType), fooMethodName, Nil)(IntType),
+      Apply(EAF, This()(thisType), barMethodName, Nil)(IntType)
+    )(IntType)
+
+    val classDefs = Seq(
+      classDef("A", superClass = Some(ObjectClass),
+        methods = List(
+          trivialCtor("A"),
+          MethodDef(EMF, mainMethodName, NON, Nil, IntType, Some(mainBody))(EOH, UNV),
+          MethodDef(EMF, fooMethodName, NON, Nil, IntType, Some(int(1)))(EOH, UNV),
+          MethodDef(EMF, barMethodName, NON, Nil, IntType, Some(int(2)))(EOH, UNV)
+        )
+      )
+    )
+
+    val requirements = {
+      reqsFactory.instantiateClass("A", NoArgConstructorName) ++
+      reqsFactory.callMethod("A", mainMethodName)
+    }
+
+    val analysisFuture = computeAnalysis(classDefs, requirements,
+        config = StandardConfig().withSemantics(_.withProductionMode(productionMode)))
+
+    for (analysis <- analysisFuture) yield {
+      assertNoError(analysis)
+
+      val AfooMethodInfo = analysis.classInfos("A")
+        .methodInfos(MemberNamespace.Public)(fooMethodName)
+      assertTrue(AfooMethodInfo.isReachable)
+
+      val AbarMethodInfo = analysis.classInfos("A")
+        .methodInfos(MemberNamespace.Public)(barMethodName)
+      assertFalse(AbarMethodInfo.isReachable)
+    }
+  }
+
+  @Test
+  def linkTimeIfError(): AsyncResult = await {
+    val mainMethodName = m("main", Nil, IntRef)
+    val fooMethodName = m("foo", Nil, IntRef)
+    val barMethodName = m("bar", Nil, IntRef)
+
+    val thisType = ClassType("A", nullable = false)
+
+    val productionMode = true
+
+    /* linkTimeIf(unknownProperty) {
+     *   this.foo()
+     * } {
+     *   this.bar()
+     * }
+     */
+    val mainBody = LinkTimeIf(
+      BinaryOp(BinaryOp.Boolean_==,
+          LinkTimeProperty("core/unknownProperty")(BooleanType),
+          BooleanLiteral(productionMode)),
+      Apply(EAF, This()(thisType), fooMethodName, Nil)(IntType),
```

Comment:
Consider:

```suggestion
      Apply(EAF, This()(thisType), barMethodName, Nil)(IntType),
```

IIUC this will both simplify and strengthen the test.